packages/labextension/src/index.ts

```diff
@@ -1,69 +1,110 @@
 
+import {
+  INotebookTracker,
+  INotebookWidgetFactory,
+  NotebookPanel,
+  NotebookWidgetFactory,
+} from "@jupyterlab/notebook";
+import { IRenderMimeRegistry } from "@jupyterlab/rendermime";
+
+import {
+  ITranslator,
+  nullTranslator,
+  TranslationBundle,
+} from "@jupyterlab/translation";
+
+import { markdownIcon } from "@jupyterlab/ui-components";
+
+interface IJupytextFormat {
+  /**
+   * Conversion format
+   */
+  format: string;
+  /**
+   * Command label
+   */
+  label: string;
+}
+
 interface JupytextRepresentation {
   format_name: string;
   extension: string;
-};
+}
 
 interface JupytextSection {
   formats?: string;
   notebook_metadata_filter?: string;
   cell_metadata_filter?: string;
-  text_representation?: JupytextRepresentation
-};
+  text_representation?: JupytextRepresentation;
+}
 
-const JUPYTEXT_FORMATS = [
-  {
-    format: "ipynb",
-    label: "Pair Notebook with ipynb document"
-  },
+function getJupytextFormats(trans: TranslationBundle): IJupytextFormat[] {
+  return [
     {
-    format: "auto:light",
-    label: "Pair Notebook with light Script"
-  },
-  {
-    format: "auto:percent",
-    label: "Pair Notebook with percent Script"
-  },
-  {
-    format: "auto:hydrogen",
-    label: "Pair Notebook with Hydrogen Script"
-  },
-  {
-    format: "auto:nomarker",
-    label: "Pair Notebook with nomarker Script"
-  },
-  {
-    format: "md",
-    label: "Pair Notebook with Markdown"
-  },
-  {
-    format: "md:myst",
-    label: "Pair Notebook with MyST Markdown"
-  },
-  {
-    format: "Rmd",
-    label: "Pair Notebook with R Markdown"
-  },
-  {
-    format: "qmd",
-    label: "Pair Notebook with Quarto (qmd)"
-  },
-  {
-    format: "custom",
-    label: "Custom pairing"
-  },
-  {
-    format: "none",
-    label: "Unpair Notebook"
-  }
-];
+      format: "ipynb",
+      label: trans.__("Pair Notebook with ipynb document"),
+    },
+    {
+      format: "auto:light",
+      label: trans.__("Pair Notebook with light Script"),
+    },
+    {
+      format: "auto:percent",
+      label: trans.__("Pair Notebook with percent Script"),
+    },
+    {
+      format: "auto:hydrogen",
+      label: trans.__("Pair Notebook with Hydrogen Script"),
+    },
+    {
+      format: "auto:nomarker",
+      label: trans.__("Pair Notebook with nomarker Script"),
+    },
+    {
+      format: "md",
+      label: trans.__("Pair Notebook with Markdown"),
+    },
+    {
+      format: "md:myst",
+      label: trans.__("Pair Notebook with MyST Markdown"),
+    },
+    {
+      format: "Rmd",
+      label: trans.__("Pair Notebook with R Markdown"),
+    },
+    {
+      format: "qmd",
+      label: trans.__("Pair Notebook with Quarto (qmd)"),
+    },
+    {
+      format: "custom",
+      label: trans.__("Custom pairing"),
+    },
+    {
+      format: "none",
+      label: trans.__("Unpair Notebook"),
+    },
+  ];
+}
+
+/**
+ * Supported file format.
+ */
+const SUPPORTED_EXTENSION = ["ipynb", "md", "Rmd", "qmd"];
```

Comment:
Thanks for factorizing this! I understand that putting a name on this is not easy... what do you think of something like `LANGUAGE_INDEPENDENT_NOTEBOOK_EXTENSIONS` ?